### ISSUE TITLE
Dexcom Share (non-us) no data issue

### ISSUE THREAD

**Issue by jannyg** [loopandlearn/LoopFollow#164]
Hi,

I have recently tested this app in testflight with no luck of getting data from Dexcom Share. I also now tried building from master and dev brancg with the same issue. I'm not a developer, but I have debugged the app and will be happy to test any changes and suggestions that anyone might have. 

The authorization seems to work fine, but this error message is being logged:

`
2022-04-12 16:15:37.618034+0200 Loop Follow[853:73036] Task <C9DCC3DB-8395-4B7D-AECF-BA5FFAB2E4FF>.<13> finished with error [-1002] Error Domain=NSURLErrorDomain Code=-1002 "unsupported URL" UserInfo={NSLocalizedDescription=unsupported URL, NSErrorFailingURLStringKey=/api/v1/entries/sgv.json?count=289, NSErrorFailingURLKey=/api/v1/entries/sgv.json?count=289, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <C9DCC3DB-8395-4B7D-AECF-BA5FFAB2E4FF>.<13>"
), _NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <C9DCC3DB-8395-4B7D-AECF-BA5FFAB2E4FF>.<13>, NSUnderlyingError=0x282734e70 {Error Domain=kCFErrorDomainCFNetwork Code=-1002 "(null)"}}
`

**Comment by jonfawcett** [comment]
This is failing on nightscout, not dex share. Without seeing more details (full xcode screenshot), my only guess is that you have a blank space in the NS url field.

**Comment by jannyg** [comment]
I'm pretty sure I didn't set any NS settings as I thought it was possible to only use the Dexcom option. I can update with screenshots later.

**Comment by jonfawcett** [comment]
What I’m saying is the error you pasted comes from the nightscout code, not the dexcom code. That code section is only run when there is text entered into the field in settings and it looks like the text is just an empty space based on the error message.

**Comment by jannyg** [comment]
I have now confirmed that NS settings was blank (only Dexcom set) and I have also verified that the same happens with the simulator. Just to be clear, is it possible to use with only Dexcom set? Because the main issue is that I don't get any data pulled from the Dexcom Share service and if NS is needed for everything to work, then that's the issue. I was hoping for the app to also be a better replacement for Dexcom follow to allow for snoozing alarms for a set time period etc. 

![Simulator Screen Shot - iPhone 12 Pro - 2022-04-13 at 08 36 55](https://user-images.githubusercontent.com/387004/163115429-e53b2a76-ec5f-474b-b4d7-d8f39093601d.png)
![Simulator Screen Shot - iPhone 12 Pro - 2022-04-13 at 08 37 00](https://user-images.githubusercontent.com/387004/163115435-56fb5462-51ca-4ef7-b256-199cb1179b30.png)
![image](https://user-images.githubusercontent.com/387004/163115395-c186a9f8-984a-4bbb-a873-6b2589e36314.png)

**Comment by jonfawcett** [comment]
Yes it does work with dexcom only. It tries to use dexcom first if you have entered the login and will try to use NS if dexcom fails. The weird thing here is that I don’t see anything in your log from dexcom before it tries to download from NS.  I looked at the code and found a few unique situations where it could be trying to use NS when it shouldn’t. The cases I found:

1. You selected to show more than 24 hours in the graph. I think Dexcom can only pull 24 hours, so NS was required for this. I changed this so that it should ignore this setting if you only have dexcom setup.
2. Dexcom returned a valid reading but it was outdated. I noticed that this would try NS even if it wasn’t set.
3. Certain errors from the attempt to download from dexcom

If you had set the graph to more than 24 hours, change that and see if it solves your issue. If you hadn’t, then grab the latest Dev code and build again to see if it works. If it still doesn’t work, I will need to see an in-process breakpoint. Open Controllers/Nightscout.swift and click Line 75 on the left of the code. This will flag it blue (a breakpoint). Then build and let it run. When xcode pauses for this line, expand err and result in the bottom left pane so I can see what is getting returned by Dexcom.

To show both of the lower panes, you need to click the 2 little squares on the bottom right of the center area of Xcode.


![IMG_0202](https://user-images.githubusercontent.com/38429455/163171453-99c93e92-6642-4b9e-b242-e45fe91679f1.jpeg)

**Comment by jannyg** [comment]
I haven't changed ant graph settings and verified it was set to 1 day. Here's the result from the test of the dev branch after deleting the app and installing it again and setting username/password for Dexcom Share only. 

<img width="1792" alt="image" src="https://user-images.githubusercontent.com/387004/163211476-6ae2450f-2e5d-4faa-a2e7-217edabeafd7.png">

**Comment by jonfawcett** [comment]
Looks like your dexcom login isn’t correct

**Comment by jannyg** [comment]
Hmm. It looks like I need to look further into this. I have been using the same login (Dexcom Clarity) as I did with NS (bridge) and I thought I could use it here as well, but it could be that the same credentials not could be used for the Share/Follow service then.

I'm a caregiver and for Dexcom Share/Follow I only received an invitation with a code that sets up the Dexcom Follow app automatically. Thus I had no actual Dexcom users, and I have tried creating a new Dexcom user with the same e-mail address as used for Follow, but no luck. I'll try to see if I can authenticate and pulls some data from the API manually somehow.
